DifferentiationInterface/ext/DifferentiationInterfaceMooncakeExt/differentiate_with.jl

```diff
@@ -0,0 +1,22 @@
+@is_primitive MinimalCtx Tuple{CoDual{<:DI.DifferentiateWith},CoDual{<:AbstractArray}}
+@is_primitive MinimalCtx Tuple{CoDual{<:DI.DifferentiateWith},CoDual{<:Number}}
+
+function Mooncake.rrule!!(dw::CoDual{<:DI.DifferentiateWith}, args::CoDual...)
```

Comment:
```suggestion
function Mooncake.rrule!!(dw::CoDual{<:DI.DifferentiateWith}, x::CoDual)
```
Right now this only works for a single argument anyway, right?

Reply:
yeah, ill change it up once its perfect for the single arg case. thanks